src/model/enhance-numbered-lists.ts

```diff
@@ -167,6 +167,65 @@ const addH3s = (elements: CAPIElement[]): CAPIElement[] => {
 	return withH3s;
 };
 
+const isFullReviewLink = (element: CAPIElement): boolean => {
+	if (!element) return false;
+	// Checks if this element is a 'full review' based on the convention: <ul> <li><p><strong>Full review:</strong> <a href="https://www.theguardian.com/technology/2019/oct/22/oneplus-7t-pro-review-the-best-kind-of-deja-vu">OnePlus 7T Pro review: the best kind of deja vu</a></p></li> </ul>
+	if (
+		element._type !== 'model.dotcomrendering.pageElements.TextBlockElement'
+	) {
+		return false;
+	}
+	const frag = JSDOM.fragment(element.html);
+	if (!frag || !frag.firstElementChild) return false;
+
+	const hasULWrapper = frag.firstElementChild.nodeName === 'UL';
+	const hasLINestedWrapper =
+		frag.firstElementChild.firstElementChild &&
+		frag.firstElementChild.firstElementChild.nodeName === 'LI';
+
+	const wrapper =
+		frag.firstElementChild.firstElementChild &&
+		frag.firstElementChild.firstElementChild.firstElementChild;
+
+	const hasPWrapper = wrapper && wrapper.nodeName === 'P';
+	const containsStrongTags =
+		wrapper && wrapper.outerHTML.includes('<strong>');
+	const containsLinks = wrapper && wrapper.outerHTML.includes('<a ');
+
+	const hasFullReviewText =
+		wrapper &&
+		wrapper.firstElementChild &&
+		wrapper.firstElementChild.textContent === 'Full review:';
+
+	return (
+		!!hasULWrapper &&
+		!!hasLINestedWrapper &&
+		!!hasPWrapper &&
+		!!containsStrongTags &&
+		!!containsLinks &&
+		!!hasFullReviewText
```

Comment:
Do you need the double bangs?

Reply:
because every value can be `boolean`, `undefined` or `null`

Reply:
I think you could remove them and it would still work? I find it's easier to read without them

Reply:
TS isnt happy
![Screenshot 2021-04-23 at 09 31 47](https://user-images.githubusercontent.com/8831403/115843453-c4079880-a416-11eb-8526-1ce0db3cfb86.png)

Reply:
Sorry to harp on herebut I'm inclined to dig into this more as it potentially signals there might be a logic issue. Seems like the issue is with `wrapper`, that is potentially `undefined` which then carries through. So what should happen if `wrapper` is undefined?

But perhaps we don't care? Maybe this logic can be simplified? Looking at FE, I see they use this

```
      // Styled link/section end
      document.select("ul > li:only-child").asScala.foreach { li =>
        val ul = li.parent();
        ul.addClass("article-link")
      }
```

So the convention there is, has UL tag with only one LI tag as a child. So maybe we don't need to examine the third child like:

```
const wrapper = frag.firstElementChild?.firstElementChild?.firstElementChild;
```

One option would be to return true when we see `<ul><li>{someHtml}</li></ul>`and then dangerously set that html? That would seem to respect the rules laid out in FE and also give us maximum flexibility. This also gets us around the problem where we're expeccting 'Full review: {someLink}'